src/components/NavSection.tsx

```diff
@@ -21,8 +22,30 @@ export const NavSection = ({
 
   const isActive = sortedNavEntries.some((entry) => entry.id === activeItem)
 
-  const items = sortedNavEntries.map((entry) => (
-    <NavEntry key={entry.id} entry={entry} isActive={activeItem === entry.id} />
+  let navItems = sortedNavEntries
+  if (sectionId === 'components') { 
+    // only display unique entry.data.id in the nav list if the section is components
+    navItems = [
+      ...sortedNavEntries
+        .reduce((map, entry) => {
+          if (!map.has(entry.data.id)) {
+            map.set(entry.data.id, entry)
+          }
+          return map
+        }, new Map())
+        .values(),
+    ]
+  }
+
+  const items = navItems.map((entry) => (
+    <NavEntry
+      key={entry.id}
+      entry={entry}
+      isActive={
+        activeItem === entry.id ||
+        window.location.pathname.includes(kebabCase(entry.data.id))
+      }
+    />
```

Comment:
Also can you update the tests for NavSection to reflect these change in behavior?